lib/evmone/instructions.hpp

```diff
@@ -93,14 +94,15 @@ inline bool check_memory(ExecutionState& state, const uint256& offset, uint64_t
         return false;
 
     const auto new_size = static_cast<uint64_t>(offset) + size;
-    if (new_size > state.memory.size())
-        state.gas_left = grow_memory(state.gas_left, state.memory, new_size);
+    if (new_size > memory.size())
+        gas_left = grow_memory(gas_left, memory, new_size);
 
-    return state.gas_left >= 0;  // Always true for no-grow case.
+    return gas_left >= 0;  // Always true for no-grow case.
 }
 
-// Check memory requirements for "copy" instructions.
-inline bool check_memory(ExecutionState& state, const uint256& offset, const uint256& size) noexcept
+/// Check memory requirements for "copy" instructions.
+inline bool check_memory(
```

Comment:
Not sure the asymmetry between `grow_memory` and `check_memory` of how `gas_left` is used is good. Should probably keep one option, pass-by-reference or return, for both.

Reply:
The `check_memory` is `inline` so it does not matter. The `grow_memory` is `noinline` so calling convention will force `int64_t& gas` to be passed via memory.

The`check_memory` can be changed too but it has much more users and the changed version wasn't looking very good. So this should be a separate piece of work. Not needed for performance though.